.github/workflows/ci.yml

```
@@ -201,5 +205,28 @@ jobs:
         uses: actions/download-artifact@v2
         with:
           name: nuget-packages
+
+      # if tag cointains suffix "-" treat it as pre-release
+      # example: 1.0.0-beta1
+      - name: Prerelease logic
+        id: prerelease-logic
+        run:
+          if [[ ${{ needs.package.outputs.tag }} =~ .+-.+$ ]]
+          ;then 
+            echo "::set-output name=prerelease::true"
+          ;else
+            echo "::set-output name=prerelease::false" 
+          ;fi
+
+      - name: Create release
+        uses: softprops/action-gh-release@v1
+        with:
+          name: Release v${{ needs.package.outputs.tag }}
+          draft: true
+          prerelease: ${{ steps.prerelease-logic.outputs.prerelease }}
```

The whole prerelease condition could probably be simplified by using [`contains`](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#contains)

---

would be nice, tho I don't see regex is supported, we could search for `-`, and if you think its good enough lets do it ⚡

---

I _think_ that searching for `-` should be enough for determining it's a prerelease.

@MoFtZ @m4rs-mt do you confirm?

---

I believe that is correct.

According to the [Nuget documentation](https://docs.microsoft.com/en-us/nuget/concepts/package-versioning#semantic-versioning-200) and the Semantic Versioning spec, [clause 9](https://semver.org/spec/v2.0.0.html#spec-item-9), says it is a hyphen.

In theory, clause 10 also talks about build metadata, which starts with +, but may contain hyphens, but it shouldn't appear in our usage.

---

@76creates I think @jgiannuzzi was referring to removing the `prerelease-logic` step, and using `contains` as part of the `softprops/action-gh-release@v1` step.

```suggestion
          prerelease: ${{ contains(needs.package.outputs.version, '-') }}
```

---

makes total sense, I failed to see that, updated

---

you read my mind @MoFtZ 😅
I'll use the suggestion feature next time 😄